lib/function.js

```diff
@@ -16,6 +16,9 @@ exports.noop = function noop() {};
  * @return {Array} names
  */
 exports.getParamNames = function getParamNames(func, cache) {
+  if (typeof func !== 'function') {
```

Comment:
I'd like to throw error if input is not function, so it won't mixed with `function() {}`.

```js
const type = typeof func;
assert(func === 'function', 'The "func" must be of type function. Received type %s', type);
```

Reply:
It sounds a good idea. 

I'll recommit with your idea.

Reply:
@dead-horse 

I've recommitted your idea and also, updated document. 
please review it.